### PR TITLE
Call IPAM cleanup using API enpoints

### PR DESCRIPTION
**Pull request by jnummelin**
Moved the IPAM cleanup loop into IPAM launcher as it's really a task for Weave adapter.

Needs: https://github.com/kontena/kontena-docker-ipam-plugin/pull/61

As soon as we have IPAM release with the cleanup endpoint we should also pin the version and not to use `latest`.

### CODE REVIEW

File: agent/lib/kontena/launchers/ipam_plugin.rb [hunk 1]
```diff
@@ -21,6 +24,26 @@ def initialize(autostart = true)
       async.ensure_image if autostart
     end
 
+    def cleanup_ipam
```
Comment (jakolehm) [inline agent/lib/kontena/launchers/ipam_plugin.rb]: Improvement but I still think that this cleanup should be a separate actor.

**Comment by jakolehm** [comment]
Should we include this to 0.17?

**Comment by jnummelin** [comment]
I think this should be part of 0.17, otherwise there's nothing that would cleanup stale address reservations.

File: agent/lib/kontena/network_adapters/ipam_cleaner.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        debug "invoking cleanup with #{addresses.size} known addresses"
+        ipam_client.cleanup_network('kontena', addresses, cleanup_index)
+      end
+    end
+
+    def collect_local_addresses
+      debug "starting to collect locally known addresses"
+      local_addresses = []
+      Docker::Container.all(all: true).each do |container|
+        local_addresses << container.overlay_cidr
```
Comment (SpComb) [inline agent/lib/kontena/network_adapters/ipam_cleaner.rb]: Prefer `container.overlay_ip`

Comment (SpComb) [inline agent/lib/kontena/network_adapters/ipam_cleaner.rb]: 1. The kontena-ipam ignores the CIDR suffix in the cleanup addresses
2. The only place in the agent that really needs the full CIDR address is the `weave attach` part
3. For #1336, the `container.overlay_cidr` may still have the old CIDR suffix. The `weave attach` logic translates it

**Pull request merged**